src/mono/wasm/features.md

```diff
@@ -2,36 +2,270 @@
+
+## Caching, Integrity
+
+Your browser could be caching various files and assets downloaded by the dotnet runtime so that the next application start will be much faster.
+
+When you deploy a new version of the application, you need to make sure that the caches in the browser and also in any HTTP proxies will not interfere.
+
+WARNING: Your server can't force the browser to ask for the file on the same URL again, if the browser thinks that the file is fresh enough.
+
+_TODO_ discuss cache busting strategies.
+
+In order to make sure that the application resources are consistent with each other, dotnet runtime will use `integrity` hash for all downloaded resources.
+
+See also [Cache-Control on headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control)
+
+See also [fetch integrity](https://developer.mozilla.org/en-US/docs/Web/API/Request/integrity)
+
+## MIME types
+
+Are `Content-Type` HTTP headers which tell the browser about the type of downloaded asset. They are necessary for correct and fast processing by the browser, but also by various caches and proxies.
+
+HTTP headers are sent by your HTTP server or proxy, which need to be properly configured.
+
+| file extension  | Content-Type |
+|---|---|
+|.wasm|application/wasm|
+|.json|application/json|
+|.js|text/javascript|
+
+See also [Content-Type on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type)
+
+## Compression
+
+Modern browsers are able to unpack files compressed for the transport. We recommend brotli compression for best results.
+
+See also [Content-Encoding on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding)
+
+## Content security policy
+
+dotnet runtime for wasm is CSP compliant starting from .Net 8, except legacy JS interop methods.
+
+In order to enable it, please set HTTP headers similar to `Content-Security-Policy: default-src 'self' 'wasm-unsafe-eval'`
+
+HTTP headers are sent by your HTTP server or proxy, which need to be properly configured.
+
+See also [CSP on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP)
+
+See also [wasm-unsafe-eval](https://github.com/WebAssembly/content-security-policy/blob/main/proposals/CSP.md#the-wasm-unsafe-eval-source-directive)
+
+## ICU
+
+Browsers don't offer full set APIs for working with localization and so we have to bring the data and the logic as part of the application.
+
+If your application doesn't need to work with locatization, you can reduce download size by `<InvariantCulture>true</InvariantCulture>`.
+
+If your application needs to work with locatization but you don't require processing speed for processing of large texts, you can reduce download size by `<HybridGlobalization>true</HybridGlobalization>`.
+It will use browser APIs instead part of ICU database. This feature is still in development.
+
+If your application needs to work with specific subset of locales, you can create your own ICU database.
```

Comment:
Here we are missing info about sharding and general logic of ICU files (choosing based on culture etc.) that is included in https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-icu-wasm.md, so I would just redirect to that link. "To learn how Globalization works when Invariant and Hybrid modes are off, see: [doc]."